include/wx/event.h

```diff
@@ -2957,14 +2960,32 @@ class WXDLLIMPEXP_CORE wxUpdateUIEvent : public wxCommandEvent
     bool GetSetEnabled() const { return m_setEnabled; }
     bool GetSetShown() const { return m_setShown; }
 
-    void Check(bool check) { m_checked = check; m_setChecked = true; }
+    void Check(bool check) { Set3StateValue(check ? wxCHK_CHECKED : wxCHK_UNCHECKED); }
+    void Set3StateValue(wxCheckBoxState check)
+    {
+        wxCHECK_RET(IsCheckable(), "IsCheckable() not enabled");
+        wxCHECK_RET(Is3State() || check != wxCHK_UNDETERMINED, "Is3State() not enabled");
+        m_3checked = check;
+        m_setChecked = true;
+    }
```

Comment:
I don't think we should allow calling `Set3StateValue()`, even with the 2 standard values, for non-3-state checkbox, this is just confusing. I.e. I'd do something like this:

```suggestion
    void Check(bool check)
    {
        wxASSERT_MSG(IsCheckable(), "Shouldn't be called if non-checkable");
        
        m_3checked = check ? wxCHK_CHECKED : wxCHK_UNCHECKED);
        m_setChecked = true;
    }
    void Set3StateValue(wxCheckBoxState check)
    {
        wxASSERT_MSG(Is3State(), "Shouldn't be called if non-3-state-checkable");
        
        m_3checked = check;
        m_setChecked = true;
    }
```

Reply:
If this were completely new code, I probably would have done it the way you suggest.  However, `wxCheckBoxBase::Set3StateValue()` explicitly allows itself to be called with the two standard values for `!Is3State()`, so I followed its design.  Do you still want me to change the `wxUpdateUIEvent` code?

Reply:
Sorry, I forgot that `wxCheckBox` function worked like this (and I still wonder why it does, but it's too late to change this, of course...). I agree that it makes sense to make the behaviour of the new function consistent with the existing one, so please disregard this comment, sorry again.